yt/visualization/plot_window.py

```diff
@@ -578,11 +578,24 @@ def set_buff_size(self, size):
         parameters
         ----------
         size : int or two element sequence of ints
-            The number of data elements in the buffer on the x and y axes.
-            If a scalar is provided,  then the buffer is assumed to be square.
+            The number of pixels in the image buffer along the x and y axes.
+            If a scalar is provided, then the buffer is assumed to be square.
+
+        .. note::
+
+            If the image buffer is accessed via the plot's frb attribute, the
+            image shape will be the *reverse* of the size specified here. For
+            example, after doing ``plot.set_buff_size((400, 800))`` you will
+            find that ``plot.frb[field].shape == (800, 400)``. This is set up
+            this way because matplotlib's imshow command displays images in
+            row-major order.
         """
         if iterable(size):
-            self.buff_size = size
+            # matplotlib displays images with shapes (m, n) with m rows
+            # and n columns. Reversing the shape here means that if I do
+            # plot.set_buff_size((nx, ny)) where will be nx pixels along
+            # x (nx columns) and ny pixels along y (ny rows).
+            self.buff_size = (size[1], size[0])
```

Comment:
Why do you need to flip the size here? This will give a strange behavior that when I do `plot.set_buff_size((400, 800))`. I will get `plot.buff_size == ((800, 400))`

Reply:
Hmm, that's true. I guess I could do the flipping at a lower level.

Reply:
Take a look at the latest version of the pull request, I think I've fixed this.